server/conn.go

```diff
@@ -1297,6 +1298,15 @@ func (cc *clientConn) writeChunks(ctx context.Context, rs ResultSet, binary bool
 	req := rs.NewRecordBatch()
 	gotColumnInfo := false
 	for {
+		failpoint.Inject("SleepInwriteChunks", func(val failpoint.Value) {
+			time.Sleep(time.Duration(val.(int)) * time.Millisecond)
+		})
+		//MaxExecDuration() return 0 if there is no limit
+		if rs.MaxExecDuration().Nanoseconds() > 0 &&
```

Comment:
It's better to start a new goroutine to monitor all processing queries periodically and kill these who exceed their max_execution_time like #10350.

Reply:
> It's better to start a new goroutine to monitor all processing queries periodically and kill these who exceed their max_execution_time like #10350.

Added a new go routine to monitor all registered result sets. refer to the new commit.